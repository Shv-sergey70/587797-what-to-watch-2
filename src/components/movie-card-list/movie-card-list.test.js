import React from 'react';
import renderer from 'react-test-renderer';
import MovieCardList from "./movie-card-list";

const films = [
  {
    id: `id-77`,
    title: `my film title 1`,
    poster: `img/my-poster1.jpg`
  },
  {
    id: `id-78`,
    title: `my film title 2`,
    poster: `img/my-poster2.jpg`
  },
  {
    id: `id-79`,
    title: `my film title 3`,
    poster: `img/my-poster3.jpg`
  }
];

test(`MovieCardList correctly renders`, () => {
  const movieCard = renderer.create(
      <MovieCardList
        films={films}
      />).toJSON();

  expect(movieCard).toMatchSnapshot();
});
