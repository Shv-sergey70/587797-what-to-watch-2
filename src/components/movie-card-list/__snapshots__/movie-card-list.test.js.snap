// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCardList correctly renders 1`] = `
Array [
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Macbeth"
        height="175"
        src="img/my-poster1.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        my film title 1
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Macbeth"
        height="175"
        src="img/my-poster2.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        my film title 2
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Macbeth"
        height="175"
        src="img/my-poster3.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        my film title 3
      </a>
    </h3>
  </article>,
]
`;
